### ISSUE TITLE
Could we ensure a specific LeetCode session?

### ISSUE THREAD

**Issue by Zackhardtoname** [rsb-23/1page-automation#2]
It'd be nice to use a dedicated LeetCode session to solve the automated problems, and another dedicated session to solve our own problems. In way, automatically solved problems wouldn't show up as solved when we actually get around to solve them.

I think it'd be extremely helpful if we could switch to a session of a specific name or ensure we are in that session before each submission ❤️.

**Comment by Zackhardtoname** [comment]
I also added features like sending an email to oneself upon failure as well as submitting a specific problem. Lmk if I should raise some PRs

**Comment by rsb-23** [comment]
❌ Adding alert notifications via email/ slack or any other medium is not in projects' scope. 
Reason : Medium of Alert is unique to each user, hence handling each case is not feasible.

Issue description on session is not clear.

Maybe these points will help -
- Each task has to be run separately.
- All scripts are standalone and only has common config to avoid repetitive code.
- A new dedicated session is created for each task with no dependency to others.

**Comment by Zackhardtoname** [comment]
This is what I am talking about
https://leetcode.com/accounts/login/?next=/session/

**Comment by rsb-23** [comment]
Understood.

This project aims to show different usecases of browser-based automation using python.
It doesn't focus particularly on automation of 1 website, hence its out of scope.

**Comment by Zackhardtoname** [comment]
Do you use the script yourself daily? If so, I imagine the session feature would be quite helpful. It sounds like you wrote this just as a demo and nobody actively uses it

**Comment by rsb-23** [comment]
- Yes, me and my friends have used it regularly for a year - not much now.
- Feel free to fork and add session feature for the community.

**Comment by Zackhardtoname** [comment]
@rsb-23 I would love to contribute. I have got really really stuck on the session change feature tho. I copied the request from the network tab in Chrome as curl, and then ran it on my MacbookPro. It works. However, the same command doesn't work if I run it on Linux (Ubuntu 22.04.4 or Arch + KDE).

So far I have found that it requires http2, but it still fails with 403 when I specify --http2 in curl. I have looked into `-v` and `--trace-ascii` but have not found anything notable. I suspect it has to do with how the connection is made to the server. If the request is clearly `curl`, the server would turn it down.
